### DISCUSSION TITLE
WINLOGBEAT LAUNCH-FAILURE

### DISCUSSION THREAD

**Discussion by ob1xxx** [Security-Onion-Solutions/securityonion#7735] [Unsupported Versions]
\
I have an successful build of SO but issues with Winlog beat. I"m running WINLOG beat on an AWS windows 2008 R2 server. Below is the message I see on the windows server. I see no information in KIbana under the filter for winlog beat. I'e re-ran the setup twice. The versions for Kibana and winlog beat                                                                                               ji

 "established connection failed because connected host has failed to respond.]
any of the configured Elasticsearch hosts. Errors: [error connecting to Elasticsearch at ht
"http://172.31.xxx.xxx:5044": dial tcp 172.31.76.247:5044: connector: A connection attempt f
arty did not properly respond after a period of time, or established connection failed because
 to respond.]"

**Comment by dougburks** [comment]
Did you follow our Beats documentation and run so-allow to allow the traffic?
https://docs.securityonion.net/en/2.3/beats.html
https://docs.securityonion.net/en/2.3/so-allow.html

**Reply by ob1xxx** [reply]
Correct. Here's the output from the Windows box. essentially failure to connect to ElasticSearch
: A connection attempt failed because the connected party did not properly respo
d connection failed because the connected host has failed to respond.]
configured Elasticsearch hosts. Errors: [error connecting to Elasticsearch at ht
.72.x.x.x.:5044": dial tcp 172.x.x.x.5044: connectex: A connection attempt f
t properly respond after a period of time, or established connection failed beca
.]
ogbeat-oss-7.17.1-windows-x86_64_real\winlogbeat-7.17.1-windows-x86_64> output.e

**Reply by dougburks** [reply]
If it's saying that it failed to connect to Elasticsearch, did you follow these steps at https://docs.securityonion.net/en/2.3/beats.html#winlogbeat?

> Disable the output.elasticsearch output.
Enable the output.logstash output and configure it to send logs to port 5044 on your management node.

**Reply by ob1xxx** [reply]
Yes, I've repeated this step. I save the config, launch service, no errors but no logs.  I also tried just running winlogbeat with the MSI. Once I try starting the service I get error messages also. What the heck.

**Reply by dougburks** [reply]
Are there any network firewalls that could be interfering with the traffic?

If you run tcpdump on your Security Onion management interface, do you see the traffic arriving on port 5044?

**Reply by ob1xxx** [reply]
I've made some progress. It appears the windows box is sending logs via winlog beats, but the filer for logstash-beats has no logs received.

**Reply by dougburks** [reply]
When you run tcpdump to see the incoming traffic, does it show the same source IP address as what you allowed via so-allow?

**Reply by ob1xxx** [reply]
Yes, actually I do. I can see the traffic from the windows box.

**Reply by dougburks** [reply]
If you run `sudo iptables -nvL |grep 172.31`, do you see a firewall rule that allows the traffic from the IP address of the Windows box?

**Comment by ob1xxx** [comment]
Yes, theres one rule for 5044 present. Whats strange is when attempting to access the SO build via the internet on the win box, the request fails now. I can ping both ways, so thats a plus. Just need to confirm winlog is receiving and get back to the SO webpage. so-allow has everything as ok so its strange.

**Reply by dougburks** [reply]
Did the Windows IP address change?

Are there any network firewalls or other devices in between that may be interfering with the traffic?

**Comment by ob1xxx** [comment]
Had to step away for another project. So I'm seeing this essentially. All network connectivity seems okay or at the least pingable. Yes, I know that's not a word. Here's at quick screenshot. 
![image](https://user-images.githubusercontent.com/20797451/170921058-3dbf9f6f-e6c0-4146-8f9f-922c04544bd4.png)

**Reply by dougburks** [reply]
Based on the screenshot, it looks like it's trying to connect to elasticsearch. Did you follow the steps in the documentation including disabling the elasticsearch output and enabling the logstash output?
https://docs.securityonion.net/en/2.3/beats.html#winlogbeat
